src/Microsoft.AspNet.Mvc.Core/ModelBinding/GenericModelBinder.cs

```diff
@@ -13,25 +13,54 @@ public class GenericModelBinder : IModelBinder
     {
         public async Task<ModelBindingResult> BindModelAsync(ModelBindingContext bindingContext)
         {
-            var binderType = ResolveBinderType(bindingContext.ModelType);
-            if (binderType != null)
+            var bindingInfo = ResolveGenericBindingInfo(bindingContext.ModelType);
+            if (bindingInfo != null)
             {
-                var binder = (IModelBinder)Activator.CreateInstance(binderType);
+                var binder = (IModelBinder)Activator.CreateInstance(bindingInfo.ModelBinderType);
                 var result = await binder.BindModelAsync(bindingContext);
 
-                var modelBindingResult = result != null ?
-                    new ModelBindingResult(result.Model, result.Key, result.IsModelSet, result.ValidationNode) :
-                    new ModelBindingResult(model: null, key: bindingContext.ModelName, isModelSet: false);
-
-                // Were able to resolve a binder type.
+                if (result != null && result.IsModelSet)
+                {
+                    // Success - propegate the values returned by the model binder.
+                    return new ModelBindingResult(result.Model, result.Key, result.IsModelSet, result.ValidationNode);
+                }
+
+                // If this is the fallback case, and we didn't bind as a top-level model, then generate a
+                // default 'empty' model and return it.
+                var isTopLevelObject = bindingContext.ModelMetadata.ContainerType == null;
+                var hasExplicitAlias = bindingContext.BinderModelName != null;
+
+                if (isTopLevelObject && (hasExplicitAlias || bindingContext.ModelName == string.Empty))
+                {
+
+                    var model = result?.Model;
+                    if (model == null && bindingInfo.UnderlyingModelType.IsArray)
+                    {
+                        model = Array.CreateInstance(bindingInfo.UnderlyingModelType.GetElementType(), 0);
+                    }
+                    else if (model == null)
+                    {
+                        model = Activator.CreateInstance(bindingInfo.UnderlyingModelType);
+                    }
```

Comment:
I don't agree at all. Creating a new interface so we can call `new` instead of `Activator.CreateInstance` is overkill.

I agree with what you're saying from a separation of concerns point of view, but I'd rather us finish the quality push and do a real refactor than keep creating things.